apps/ui/BaseWindow.cpp

```diff
@@ -300,8 +298,30 @@ void BaseWindow::display()
 
         if (buffer)
         {
-            glDrawPixels(renderOutput.frameSize.x(), renderOutput.frameSize.y(),
-                         format, type, buffer);
+            glMatrixMode(GL_PROJECTION);
+            glLoadIdentity();
+            glOrtho(0.0f, windowSize.x(), 0.0f, windowSize.y(), -1.0f, 1.0f);
+
+            glMatrixMode(GL_MODELVIEW);
+            glLoadIdentity();
+            glViewport(0, 0, windowSize.x(), windowSize.y());
+
+            glBindTexture(GL_TEXTURE_2D, _fbTexture);
+            glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, renderOutput.frameSize.x(),
+                         renderOutput.frameSize.y(), 0, GL_RGBA, type, buffer);
+
+            glBegin(GL_QUADS);
+            glTexCoord2f(0.f, 0.f);
+            glVertex3f(0, 0, 0);
+            glTexCoord2f(0.f, 1.f);
+            glVertex3f(0, windowSize.y(), 0);
+            glTexCoord2f(1.f, 1.f);
+            glVertex3f(windowSize.x(), windowSize.y(), 0);
+            glTexCoord2f(1.f, 0.f);
+            glVertex3f(windowSize.x(), 0, 0);
+            glEnd();
+
+            glBindTexture(GL_TEXTURE_2D, 0);
```

Comment:
Maybe it's asking too much, but can you avoid using non GL3 core functions? You'll need shaders, but they are quite trivial.

Reply:
does this really help? I bluntly took what the ospray example viewer is doing, so no efforts spend so far. GL is dead anyways, so using legacy GL is not too much of a difference to GL3.